sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -819,13 +825,148 @@ private void SetNameFieldsIfNull()
             bool rangeGetContentHash,
             bool async,
             CancellationToken cancellationToken)
+        {
+            Response<BlobDownloadStreamingResult> response = await DownloadStreamingInternal(
+                range,
+                conditions,
+                rangeGetContentHash,
+                $"{nameof(BlobBaseClient)}.{nameof(Download)}",
+                async,
+                cancellationToken).ConfigureAwait(false);
+            BlobDownloadStreamingResult blobDownloadStreamingResult = response.Value;
+            BlobDownloadDetails blobDownloadDetails = blobDownloadStreamingResult.Details;
+            return Response.FromValue(
+                new BlobDownloadInfo()
+                {
+                    Content = blobDownloadStreamingResult.Content,
+                    Details = blobDownloadDetails,
```

Comment:
So I guess this means people using the old API or Query will see duplication of things like BlobType?  It's unfortunate, but I can live with that.

Reply:
This or inventing new types for new apis. I think this is better than adding another model class (and naming it)

Reply:
We could however hide props in BlobDownloadInfo with editor browsable.